src/control/XournalMain.cpp

```diff
@@ -329,12 +326,9 @@ auto exportPdf(const char* input, const char* output, const char* range, ExportB
 
     if (range) {
         // Parse the range
-        PageRangeVector exportRange = PageRange::parse(range, doc->getPageCount());
+        auto exportRange = PageRange::parse(range, doc->getPageCount());
```

Comment:
Please make this variable `exportRanges`. As it is a vector of ranges and you use `auto` it should be straightforward, that these might be multiple ranges.

Reply:
Please also check all other places. I did see this multiple times throughout your code.

Reply:
But that would be a variable names change; should that be a different PR?

Reply:
No, you may of course add any change into this PR that is connected to the intent of your PR. So any renaming is okay, as long as it has something to do with your actual modifications to the behavior of the code. Here, the usage of auto instead of the actual type removes the implicit knowledge that this is a vector of ranges. Hence, renaming the variable is crucial for good coding as otherwise, one might think this is a single range.

Also with a PR like yours that is not really big, it is a good idea to fix such problems straight away instead of opening another PR. If your PR would be covering complex changes in lets say 20 files, I would agree that you should do this in another PR.

Reply:
But then I will change `exportRange` (for example) to `exportRanges`, even though there will still be versions of it with `exportRange`. Wouldn't it be better to just replace all occurences? However, that means I'm changing files that I haven't changed in this PR.

Reply:
You will have to add all relevant files. But as you modified the return type of `PageRange::parse`, all files you must make additional changes to, should already be part of this PR.

We don't really care about minimum touched files in a PR. All we care about is that the intent of the PR matches with the changes. Because the intent of your PR is to update and fix the class `PageRange` any modification to the codebase that is in relation to this class is completely valid.

Reply:
OK I will update those occurrences to match the plural form of the name.

Reply:
I am not very sure about this...

You mentioned something about `auto`.  What if I replace `auto` by `std::vector<PageRangeEntry>`, will it then be enough? (I.e. can I leave the variable names unchanged?)

Reply:
Then there is still the issue with confusing variable names. You should make sure that within your code (and for the purpose of this PR within the whole class `PageRange`) there are no confusing variable names. Even better would be if you clean up all code that is using your code.

Reply:
You can do it in a separate commit, for now, so we can look at the impact on the codebase.